test/src/us/kbase/kbasefeaturevalues/test/KBaseFeatureValuesServerTest.java

```diff
@@ -195,61 +197,111 @@ public void testMainPipeline() throws Exception {
-        impl.estimateK(ekp, token, getContext());
-        ObjectData res1 = wscl.getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
-                .withName(estimObjName))).get(0);
-        EstimateKResult estKRes = res1.getData().asClassInstance(EstimateKResult.class);
+        EstimateKResult estKRes = impl.estimateK(ekp, token, getContext());
+
         long k = estKRes.getBestK();
+
+        /*
+        this is testing the object returned by the method
+        comparing the two fields that are populated: best K and K clusters
+        and empty additionalProperties */
         Assert.assertNotNull("k exists ", k);
         Assert.assertNotNull("c size exists ", estKRes.getEstimateClusterSizes().size());
-	//System.err.println("k "+k);
-        //System.err.println("cluster size "+estKRes.getEstimateClusterSizes().size());
         Assert.assertEquals(3, k);
         Assert.assertEquals(5, estKRes.getEstimateClusterSizes().size());
+        Map<String, Object> additionalProperties = new HashMap<String, Object>();
+        Assert.assertEquals(additionalProperties, estKRes.getAdditionalProperties());
         for (int i = 0; i < estKRes.getEstimateClusterSizes().size(); i++) {
-            Tuple2 <Long, Double> item = estKRes.getEstimateClusterSizes().get(i);
-            Assert.assertEquals(2L + i, (long)item.getE1());
-            Assert.assertTrue((double)item.getE2() > 0);
+            Tuple2<Long, Double> item = estKRes.getEstimateClusterSizes().get(i);
+            Assert.assertEquals(2L + i, (long) item.getE1());
+            Assert.assertTrue((double) item.getE2() > 0);
         }
-        impl.estimateKNew(new EstimateKParamsNew().withInputMatrix(testWsName + "/" +
+
+        //loading saved result object
+        ObjectData res1 = getWsClient().getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
+                .withName(estimObjName))).get(0);
+        EstimateKResult estKResLoad = res1.getData().asClassInstance(EstimateKResult.class);
+        long kLoad = estKResLoad.getBestK();
+        /*
+        this is testing the result object loaded from ws
+        comparing the two fields that are populated: best K and K clusters
+        and empty additionalProperties */
+        Assert.assertNotNull("k exists ", kLoad);
+        Assert.assertNotNull("c size exists ", estKResLoad.getEstimateClusterSizes().size());
+        Assert.assertEquals(3, kLoad);
+        Assert.assertEquals(5, estKResLoad.getEstimateClusterSizes().size());
+        Map<String, Object> additionalPropertiesLoad = new HashMap<String, Object>();
+        Assert.assertEquals(additionalPropertiesLoad, estKRes.getAdditionalProperties());
+        for (int i = 0; i < estKResLoad.getEstimateClusterSizes().size(); i++) {
+            Tuple2<Long, Double> item = estKResLoad.getEstimateClusterSizes().get(i);
+            Assert.assertEquals(2L + i, (long) item.getE1());
+            Assert.assertTrue((double) item.getE2() > 0);
```

Comment:
Asserting that a value is > 0 doesn't seem like a very stringent test, but it's exactly what the test was previously, so it's no worse than pre-PR anyway.